pilot/pkg/security/authz/model/generator.go

```diff
@@ -180,33 +181,33 @@ func (srcNamespaceGenerator) principal(_, value string, _ bool, useAuthenticated
 	return principalAuthenticated(m, useAuthenticated), nil
 }
 
-type srcServiceAccountGenerator struct{}
+type srcServiceAccountGenerator struct {
+	policyName types.NamespacedName
+}
 
-func (srcServiceAccountGenerator) permission(_, _ string, _ bool) (*rbacpb.Permission, error) {
+func (g srcServiceAccountGenerator) permission(_, _ string, _ bool) (*rbacpb.Permission, error) {
 	return nil, fmt.Errorf("unimplemented")
 }
 
-func (srcServiceAccountGenerator) principal(_, value string, _ bool, useAuthenticated bool) (*rbacpb.Principal, error) {
-	regex, err := serviceAccountRegex(value)
-	if err != nil {
-		return nil, err
-	}
+func (g srcServiceAccountGenerator) principal(_, value string, _ bool, useAuthenticated bool) (*rbacpb.Principal, error) {
+	regex := serviceAccountRegex(g.policyName.Namespace, value)
 	m := matcher.StringMatcherRegex(regex)
 	return principalAuthenticated(m, useAuthenticated), nil
 }
 
-func serviceAccountRegex(value string) (string, error) {
+func serviceAccountRegex(defaultNamespace string, value string) string {
 	ns, sa, ok := strings.Cut(value, "/")
 	if !ok {
-		return "", fmt.Errorf("invalid value: %v", value)
+		ns = defaultNamespace
+		sa = value
 	}
 	// Format should follow...
 	// 'spiffe://' then a trust domain + arbitrary k/v pairs
 	// '/ns/<namespace>/'
 	// optional arbitrary k/v pairs
 	// '/sa/<serviceAccount>'
 	// Either end of string OR / + arbitrary k/v pairs (the / ensures we do not match <service account>-some-junk)
-	return fmt.Sprintf("spiffe://.+/ns/%s/(.+/|)sa/%s(/.+)?", ns, sa), nil
+	return fmt.Sprintf("spiffe://.+/ns/%s/(.+/|)sa/%s(/.+)?", ns, sa)
```

Comment:
Does this enforce that value is not empty?

Reply:
No, but there is no other generators that seem to do so. The inputs here are pre-validated (already run through CheckServiceAccount)